scripts-2.0/components/TimeSelectors/index.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import './styles.scss';
+import {useEffect, useState} from "react";
+import TimePicker from "../common/TimePicker";
+import translate from "../../Translator";
+import DateTime from "../../helpers/DateTime";
+import TimeFormatter from "../../TimeFormatter";
+import Logger from "../../Logger";
+
+const dateTime = new DateTime();
+const timeFormatter = new TimeFormatter();
+const logger = new Logger();
+
+export interface TimeSelectorsInterface {
+    is12hFormat: boolean,
+    clearFormTrigger: boolean,
+    startTime: Date|null,
+    stopTime: Date|null,
+    durationFormat?: number|null,
+    onStartTimeValueChange(newValue: any): any
+    onStopTimeValueChange(newValue: any): any
+}
+
+const TimeSelectors: React.FC<TimeSelectorsInterface> = (props) => {
+    const [startTime, setStartTime] = useState<Date|null>(props.startTime);
+    const [stopTime, setStopTime] = useState<Date|null>(props.stopTime);
+    const [duration, setDuration] = useState<string>();
+    const [durationFormat, setDurationFormat] = useState<number>(timeFormatter.DEFAULT_FORMAT);
+    const [is12hFormat, setIs12hFormat] = useState<boolean>(false);
+
+    useEffect(() => {
+        setIs12hFormat(props.is12hFormat);
+
+        const interval = setInterval(() => {
+            if (stopTime == undefined) {
+                calculateDuration();
+            }
+        }, 1000);
+        return () => clearInterval(interval);
+    }, [props]);
+
+    useEffect(() => {
+        if (props.durationFormat != null) {
+            setDurationFormat(props.durationFormat);
+        }
+    }, [props.durationFormat]);
+
+    useEffect(() => {
+        if (durationFormat != null) {
+            calculateDuration();
+        }
+    }, [durationFormat]);
+
+    useEffect(() => {
+        clearForm();
+    }, [props.clearFormTrigger]);
+
+    useEffect(() => {
+        props.onStartTimeValueChange(startTime);
+        validateBoth();
+        calculateDuration();
+    }, [startTime]);
+
+    useEffect(() => {
+        props.onStopTimeValueChange(stopTime);
+        validateBoth();
+        validateStartTime();
+        calculateDuration();
+    }, [stopTime]);
+
+    const clearForm = () => {
+        setStopTime(null);
+        setStartTime(dateTime.getNowDateForDuration());
+    }
+
+    const validateBoth = () => {
+        //set stop time same as start time if stopTime is bigger
+        if (
+            startTime != undefined &&
```

Comment:
a tu czemu nie strict? to chyba to samo co != false. możemy ten warunek jakoś uszczegółowić?